gfortran 10.1 requirement not flagged by configure step

Head of dev branch (commit 8a9aa7306....)
Building on stock ubuntu 19.10.  Output is as follows:

```
gfortran -fallow-argument-mismatch -fopenmp -msse4.2 -I/usr/lib/x86_64-linux-gnu/openmpi/include -I/usr/lib/x86_64-linux-gnu/ope
nmpi/lib -I/usr/lib/x86_64-linux-gnu/openmpi/lib -DGEOPM_ENABLE_MPI  -O3   -c -o examples/fft/nas_ft-ft.o `test -f 'examples/fft
/ft.f90' || echo '../../repos/geopm/'`examples/fft/ft.f90
gfortran: error: unrecognized command line option ‘-fallow-argument-mismatch’; did you mean ‘-Wno-argument-mismatch’?
make[1]: *** [Makefile:9131: examples/fft/nas_ft-ft.o] Error 1
make[1]: Leaving directory '/home/rountree/var/build/geopm'
make: *** [Makefile:3852: all] Error 2
```

The gfortran 10.1 release notes [state:](https://groups.google.com/forum/#!topic/comp.lang.fortran/Ed8Mccy9zo8)
_Mismatches between actual and dummy argument lists in a single file are 
  now rejected with an error. Use the new option **-fallow-argument-mismatch** 
  to turn these errors into warnings; this option is implied with 
  -std=legacy.  -Wargument-mismatch has been removed._ 

Fails on gfortran 9.2.1.

--- a/examples/fft/Makefile.mk
+++ b/examples/fft/Makefile.mk
@@ -49,10 +49,10 @@ if ENABLE_OPENMP
 
     examples_fft_nas_ft_LDADD = libgeopm.la libgeopmfort.la $(MPI_FCLIBS) $(MPI_CXXLIBS)
     examples_fft_nas_ft_LDFLAGS = $(AM_LDFLAGS) $(MPI_LDFLAGS) $(OPENMP_CFLAGS)
-    examples_fft_nas_ft_FCFLAGS = -fallow-argument-mismatch -fopenmp -msse4.2 $(MPI_FCFLAGS) $(OPENMP_CFLAGS) -O3
+    examples_fft_nas_ft_FCFLAGS = -std=legacy -fopenmp -msse4.2 $(MPI_FCFLAGS) $(OPENMP_CFLAGS) -O3
     examples_fft_nas_ft_FFLAGS =  -fopenmp -msse4.2 $(MPI_FFLAGS) $(OPENMP_CFLAGS) -O3
 if HAVE_IFORT
-    examples_fft_nas_ft_FCFLAGS += -fallow-argument-mismatch -xAVX -shared-intel -mcmodel=medium -fpic
+    examples_fft_nas_ft_FCFLAGS += -std=legacy -xAVX -shared-intel -mcmodel=medium -fpic
     examples_fft_nas_ft_FFLAGS += -xAVX -shared-intel -mcmodel=medium -fpic
 endif
 endif

